python/pyarrow/tests/test_extension_type.py

```diff
@@ -1926,3 +1926,50 @@ def test_bool8_scalar():
     assert pa.scalar(1, type=pa.bool8()).as_py() is True
     assert pa.scalar(2, type=pa.bool8()).as_py() is True
     assert pa.scalar(None, type=pa.bool8()).as_py() is None
+
+
+@pytest.mark.parametrize("storage_type", (
+    pa.utf8(), pa.large_utf8(), pa.string_view(), pa.string(), pa.large_string()))
+def test_json(storage_type, pickle_module):
+    data = ['{"a": 1}', '{"b": 2}', None]
+    storage = pa.array(data, type=storage_type)
+    json_type = pa.json_(storage_type)
+    json_arr_class = json_type.__arrow_ext_class__()
+
+    assert pa.json_() == pa.json_(pa.utf8())
```

Comment:
So this means that the storage type is not taken into account for checking equality of the JSON extension type? Should it?

Reply:
I meant to test that `pa.json_()` sets storage to `pa.utf8()` by default and I'd keep this test. However as noted in https://github.com/apache/arrow/pull/13901#pullrequestreview-2324294761 disregarding storage type here is not desirable.

Reply:
> I meant to test that `pa.json_()` sets storage to `pa.utf8()` by default and I'd keep this test.

Ah, yes of course, that's good to test